File: wallendar-app/app/src/main/java/com/td/wallendar/service/GroupsService.java
package com.td.wallendar.service;

import com.td.wallendar.dtos.request.AddGroupRequest;
import com.td.wallendar.dtos.request.AddMemberByEmailRequest;
import com.td.wallendar.dtos.request.AddMembersRequest;
import com.td.wallendar.dtos.request.AddPaymentRequest;
import com.td.wallendar.dtos.response.GroupResponse;

import java.util.List;

import io.reactivex.Single;
import retrofit2.Response;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface GroupsService {
    @GET("/api/groups/allFromUser/{userId}")
    Single<Response<List<GroupResponse>>> getGroupsByUserId(@Path("userId") final long userId);

    @GET("/api/groups/{groupId}")
    Single<Response<GroupResponse>> getGroupById(@Path("groupId") final long groupId);

    @POST("/api/groups/create")
    Single<Response<GroupResponse>> createGroup(@Body AddGroupRequest addGroupRequest);

    @POST("/api/groups/{groupId}/addMemberByEmail")
    Single<Response<GroupResponse>> addMemberByEmail(@Path("groupId") long groupId, @Body AddMemberByEmailRequest addMemberByEmailRequest);

    @POST("/api/groups/{groupId}/addMembers")
    Single<Response<GroupResponse>> addMembers(@Path("groupId") final long groupId, @Body AddMembersRequest addMembersRequest);

    @POST("/api/groups/{groupId}/addPayment")
    Single<Response<GroupResponse>> addPayment(@Path("groupId") final long groupId, @Body AddPaymentRequest addPaymentRequest);
}

File: wallendar-app/app/src/main/java/com/td/wallendar/service/GroupsService.kt
package com.td.wallendar.service

import com.td.wallendar.dtos.request.AddGroupRequest
import com.td.wallendar.dtos.request.AddMemberByEmailRequest
import com.td.wallendar.dtos.request.AddMembersRequest
import com.td.wallendar.dtos.request.AddPaymentRequest
import com.td.wallendar.dtos.response.GroupResponse
import io.reactivex.Single
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface GroupsService {
    @GET("/api/groups/allFromUser/{userId}")
    fun getGroupsByUserId(@Path("userId") userId: Long): Single<Response<MutableList<GroupResponse>>>
    @GET("/api/groups/{groupId}")
    fun getGroupById(@Path("groupId") groupId: Long): Single<Response<GroupResponse>>
    @POST("/api/groups/create")
    fun createGroup(@Body addGroupRequest: AddGroupRequest): Single<Response<GroupResponse>>
    @POST("/api/groups/{groupId}/addMemberByEmail")
    fun addMemberByEmail(@Path("groupId") groupId: Long, @Body addMemberByEmailRequest: AddMemberByEmailRequest): Single<Response<GroupResponse>>
    @POST("/api/groups/{groupId}/addMembers")
    fun addMembers(@Path("groupId") groupId: Long, @Body addMembersRequest: AddMembersRequest): Single<Response<Void>>
    @POST("/api/groups/{groupId}/addPayment")
    fun addPayment(@Path("groupId") groupId: Long, @Body addPaymentRequest: AddPaymentRequest): Single<Response<Void>>
}